ethpm_types/contract_type.py

```diff
@@ -257,54 +258,102 @@ def parse(self) -> Dict[int, PCMapItem]:
         return results
 
 
-class ABIList(list):
+T = TypeVar("T", bound=Union[MethodABI, EventABI])
```

Comment:
Would this work?
```suggestion
T = TypeVar("T", bound=ABI)
```

https://github.com/ApeWorX/ethpm-types/blob/6b320075b7fbb796738cd851ba14c3bde361d5f6/ethpm_types/abi.py#L457-L466

Reply:
Not exactly, because it needs to additional constraint of having `.selector` property, which the one-off ABIs classes don't have